### PR TITLE
Run scorecards only on pushes to main.

### PR DESCRIPTION
**Pull request by godofredoc**
This is to prevent the workflow from failing when running in branches
with no previous data to compare.

It also adds the scorecard badge to the README file.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [X] I signed the [CLA].
- [X] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [X] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: .github/workflows/scorecards-analysis.yml [hunk 1]
```diff
@@ -1,9 +1,8 @@
 name: Scorecards supply-chain security
 on:
-  # Only the default branch is supported.
-  branch_protection_rule:
   push:
-    branches: [ main ]
+    branches:
+    - main
```
Comment (stuartmorgan) [inline .github/workflows/scorecards-analysis.yml:5]: Aren't these just different syntax for the same thing? Or is the fix the removal of the empty entry above?

Comment (godofredoc) [reply .github/workflows/scorecards-analysis.yml:5]: Yeah not sure it the empty branch protection was causing this workflow to run on push of any branch.

**Review by stuartmorgan — APPROVED** [review]
LGTM I guess, it's just not clear to me that this will actually fix it.

If it does, we need the same change in flutter/plugins.

**Review by ditman — APPROVED** [review]
Odd that we haven't seen more often. Dependabot creates branches all the time that should be triggering this more often? (maybe?)

**Comment by godofredoc** [comment]
> Odd that we haven't seen this more often. Dependabot creates branches all the time that should be triggering this more often? (maybe?)
> 
> (Also, this will have to be merged with the submit-queue in red to attempt to fix it)

Yeah but dependabot is actually creating a PR in this case it was a direct push, should I disable that to make packages consistent with flutter/flutter and flutter/engine?

**Comment by stuartmorgan** [comment]
I'm not aware of a reason we should have a different setup here than flutter/flutter and flutter/engine, so I'd be fine with aligning the configurations and seeing if anything breaks.

**Comment by ditman** [comment]
> dependabot is actually creating a PR in this case it was a direct push, should I disable that to make packages consistent with flutter/flutter and flutter/engine?

No, I meant that dependabot creates branches in the repo to create their PRs, similar to the branch that was causing the error?

Anyway, the Scorecard is now passing on master, yay!

**Pull request merged**